daemon/policy_test.go

```diff
@@ -488,3 +488,45 @@ func (ds *DaemonSuite) TestRemovePolicy(c *C) {
 	networkPolicies = ds.getXDSNetworkPolicies(c, nil)
 	c.Assert(networkPolicies, HasLen, 0)
 }
+
+func (ds *DaemonSuite) TestPolicyAddQueue(c *C) {
+	lblBar := labels.ParseLabel("bar")
+	lblFoo := labels.ParseLabel("bar")
+	lbls := labels.ParseLabelArray("foo", "bar")
+	originalRules := api.Rules{
+		{
+			Labels:           lbls,
+			EndpointSelector: api.NewESFromLabels(lblBar),
+			Egress:           []api.EgressRule{{ToCIDR: []api.CIDR{"1.1.1.1/32", "2.2.2.0/24"}}},
+		},
+	}
+
+	newRules := originalRules.DeepCopy()
+	newRules[0].EndpointSelector = api.NewESFromLabels(lblFoo)
+
+	go ds.d.PolicyQueueWorker()
```

Comment:
It would be good to stop this when the test exits. It's a little complicated because we're reusing ds.d but this will cause weird interactions between tests when we have another test using this (which is all of them now).